erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```
@@ -2097,6 +2099,112 @@ def set_status(self, update=False, status=None, update_modified=True):
+		for update in updates:
+			# Get current billed_amt to avoid overwriting other invoices' contributions
+			current_billed_amt = frappe.db.get_value("Sales Order Item", update["name"], "billed_amt") or 0
+			
+			# Add/subtract this invoice's contribution using original rate
+			new_billed_amt = flt(current_billed_amt) + flt(update["original_rate_amount"])
+			
+			frappe.db.set_value("Sales Order Item", update["name"], "billed_amt", new_billed_amt)
+
+		# Trigger recalculation of percentages and status
+		frappe.db.sql("""
+			UPDATE `tabSales Order` 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)